pkg/constraintapi/lua/acquire.lua

```diff
@@ -259,75 +259,106 @@ end
 ---@param period_ms integer
 ---@param limit integer
 ---@param burst integer
----@return integer[]
-local function throttleCapacity(key, now_ms, period_ms, limit, burst)
-	-- TODO: Reuse shared script
+---@param quantity integer
+local function throttle(key, now_ms, period_ms, limit, burst, quantity)
+	---@type { allowed: boolean, limit: integer?, retry_after: integer?, reset_after: integer?, remaining: integer? }
+	local result = {}
 
-	-- calculate emission interval (tau) - time between each token
+	-- limit defines the maximum number of requests that can be admitted at once (irrespective of current usage)
+	result["limit"] = burst + 1
+
+	-- emission defines the window size
 	local emission = period_ms / math.max(limit, 1)
+	result["ei"] = emission
 
-	-- calculate total capacity in time units
-	local total_capacity_time = emission * (limit + burst)
+	-- retry_at is always computed under the assumption that all
+	-- remaining capacity is consumed
+	result["retry_at"] = now_ms + emission
 
-	-- retrieve theoretical arrival time
-	local tat = call("GET", key)
+	-- dvt determines how many requests can be admitted
+	local dvt = emission * (burst + 1)
+	result["dvt"] = dvt
+
+	-- use existing tat or start at now_ms
+	local tat = redis.call("GET", key)
 	if not tat then
 		tat = now_ms
 	else
 		tat = tonumber(tat)
 	end
 
-	-- remaining capacity in time units
-	local time_capacity_remain = now_ms + total_capacity_time - tat
+	result["tat"] = tat
 
-	-- Convert the remaining time budget back into a number of tokens.
-	local capacity = math.floor(time_capacity_remain / emission)
+	local origQuantity = quantity
```

Comment:
worth copying over the comment from throttled?

https://github.com/throttled/throttled/blob/d5ee505c72a471702dfb7eca8e156d592bf9a68c/rate.go#L210